### ISSUE TITLE
Feature Request: Add Custom Achievements and Rewards

### ISSUE THREAD

**Issue by segsmachine** [Zachatoo/obsidian-achievements#8]
**Is your feature request related to a problem? Please describe.**
The achievements are an amazing idea. However, since I already know my way around Obsidian, the achievements are rendered useless. In addition, the lack of rewards feels unsatisfying.

**Describe the solution you'd like**
Adding custom achievements with the option to create new ones. It can extract achievements from a markdown file in the vault. In addition, adding a rewards option similar to https://github.com/Gnopps/obsidian-rewarder

Ideally, the earned rewards would be added to an inventory

**Describe alternatives you've considered**
Rewarder is a great plugin, but it's also quite buggy. In addition, my rewards aren't tracked

**Additional context**
Gamifying productivity can help increase motivation, especially with individuals who have Autism, ADHD, or general focus/motivation issues. There is currently no plugins that add this functionality to Obsidian

**Comment by Zachatoo** [comment]
This is a neat idea. It sounds like obsidian-rewarder already mostly does what you want, just needs some updates. I think it would be more work to add this feature to this plugin that it would be to update that existing plugin.

I'm open to adding it to this plugin, just not sure if it's within the scope of this plugin when there's an alternative plugin that basically does what you want.

I definitely see where you're coming from as far as the achievements not being very useful for experienced users. My target audience was newer users to Obsidian initially, then over time adding more unique achievements that would slowly start catering to more experienced users.

**Comment by segsmachine** [comment]
> obsidian-rewarder already mostly does what you want

Yes, that is true. However, the plugin has been abandoned for 8 months and bugs are piling up. I understand that this is a pretty large scope and the adding more advanced achievements is a great idea. Allowing for custom ones would be easier in the long run

The plugin is great for beginners. It's straightforward and right to the point. Adding more configuration while still having default achievements would definitely help noobs understand how plugin customization works

**Comment by franz-che** [comment]
I, too, would like to have the ability to create my own achievements. It doesn't have to include the reward system since I can just track them on my own every time I complete an achievement. Also, obsidian-rewarder doesn't track progress, you rely on ticking checkboxes.

**Comment by Zachatoo** [comment]
How would you expect the experience to be like making progress on custom achievements? Is it updating some numbers in a markdown note? In the achievements pane? Checkboxes?

**Comment by segsmachine** [comment]
> How would you expect the experience to be like making progress on custom achievements? Is it updating some numbers in a markdown note? In the achievements pane? Checkboxes?

I'd envision custom achievement progress as using sub tasks in a task list

**Comment by segsmachine** [comment]
Missclicked, my apologies
